src/libraries/Microsoft.Extensions.Configuration/tests/ConfigTests.cs

```diff
@@ -0,0 +1,1200 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Extensions.Configuration.Memory;
+using Microsoft.Extensions.Primitives;
+using Moq;
+using Xunit;
+
+namespace Microsoft.Extensions.Configuration.Test
+{
+    public class ConfigTests
+    {
+        [Fact]
+        public void AutoUpdates()
+        {
+            var config = new Config();
+
+            config.AddInMemoryCollection(new Dictionary<string, string>
+            {
+                { "TestKey", "TestValue" },
+            });
+
+            Assert.Equal("TestValue", config["TestKey"]);
+        }
+
+        [Fact]
+        public void TriggersReloadTokenOnSourceAddition()
+        {
+            var config = new Config();
+
+            var reloadToken = ((IConfiguration)config).GetReloadToken();
+
+            Assert.False(reloadToken.HasChanged);
+
+            config.AddInMemoryCollection(new Dictionary<string, string>
+            {
+                { "TestKey", "TestValue" },
+            });
+
+            Assert.True(reloadToken.HasChanged);
+        }
+
+
+        [Fact]
+        public void SettingValuesWorksWithoutManuallyAddingSource()
+        {
+            var config = new Config
+            {
+                ["TestKey"] = "TestValue",
+            };
+
+            Assert.Equal("TestValue", config["TestKey"]);
+        }
+
+        [Fact]
+        public void SettingConfigValuesDoesNotTriggerReloadToken()
+        {
+            var config = new Config();
+            var reloadToken = ((IConfiguration)config).GetReloadToken();
+
+            config["TestKey"] = "TestValue";
+
+            Assert.Equal("TestValue", config["TestKey"]);
+
+            // ConfigurationRoot doesn't fire the token today when the setter is called. Maybe we should change that.
```

Comment:
Should this be an issue rather than a comment?

Reply:
I think consistency with ConfigurationRoot is more important than this unusual use case. I would probably support the idea if people start asking for it, but I'm going to just delete the comment for now.